### PR TITLE
ELY-2616 : Replaced the switch statement in VaultObjectInputStream with an if/else statement

### PR DESCRIPTION
**Pull request by BrindaPuri**
Issue : https://issues.redhat.com/browse/ELY-2616?filter=12383825

### CODE REVIEW

**Comment by SoniaZaldana** [comment]
Hi @BrindaPuri, thanks for your contribution! Could you link the issue in the PR description?

File: credential/store/src/main/java/org/wildfly/security/credential/store/impl/VaultObjectInputStream.java [hunk 1]
```diff
@@ -31,9 +31,9 @@ final class VaultObjectInputStream extends ObjectInputStream {
 
     protected Class<?> resolveClass(final ObjectStreamClass desc) throws IOException, ClassNotFoundException {
         final String name = desc.getName();
-        switch (name) {
-            case SecurityVaultData.PICKETBOX_CLASS_NAME: return SecurityVaultData.class;
-            default: return super.resolveClass(desc);
+        if (name.equals(SecurityVaultData.PICKETBOX_CLASS_NAME)) {
+            return SecurityVaultData.class;
         }
+        return super.resolveClass(desc);
```
Comment (ashley-abdelsayed98) [inline credential/store/src/main/java/org/wildfly/security/credential/store/impl/VaultObjectInputStream.java:37]: we can actually simplify this further by using a ternary statement:
```suggestion
        return name.equals(SecurityVaultData.PICKETBOX_CLASS_NAME) ?  SecurityVaultData.class : super.resolveClass(desc);
```

File: credential/store/src/main/java/org/wildfly/security/credential/store/impl/VaultObjectInputStream.java [hunk 2]
```diff
@@ -31,9 +31,9 @@ final class VaultObjectInputStream extends ObjectInputStream {
...
     }
-}
+}
```
Comment (ashley-abdelsayed98) [inline credential/store/src/main/java/org/wildfly/security/credential/store/impl/VaultObjectInputStream.java:39]: nit pick: missing new line at end of file

File: credential/store/src/main/java/org/wildfly/security/credential/store/impl/VaultObjectInputStream.java (hunk 1 above)
Comment (ashley-abdelsayed98) [reply credential/store/src/main/java/org/wildfly/security/credential/store/impl/VaultObjectInputStream.java:37]: *edited above to exclude the if statement

**Comment by skyllarr** [comment]
@BrindaPuri Just a minor, please include an issue title in the commit title. You can change the title with `git commit --amend`

**Comment by skyllarr** [comment]
@BrindaPuri If you need any help fixing this PR, just let us know and we will gladly help! Thank you!

**Comment by skyllarr** [comment]
@BrindaPuri I am so sorry, it seems that this issue was not assigned to you in jira, so another participants worked on it and it is now superseded by https://github.com/wildfly-security/wildfly-elytron/pull/1968 , so I have to close this PR. Let us know if you need any help with creating a jira account and with choosing another issue to work on. Thank you!

**Pull request closed without merging**